neps/nep-0452.md

```diff
@@ -0,0 +1,213 @@
+- calls send on the contract and passes in the pubKey1 as an argument as well as the desired balance for the linkdrop.
+- The contract will map the `pubKey1` to the desired balance for the linkdrop.
+- The contract will then add the `pubKey1` as a function call access key with the ability to call claim and create_account_and_claim. This means that anyone with the `privKey1` that was created locally, can claim this linkdrop.
+- Linkdrop creator will then create a link to send to someone that contains this privKey1. The link may follow the format: wallet.testnet.near.org/linkdrop/{fundingContractAccountId}/{linkdropKeyPairSecretKey}?redirectUrl={redirectUrl}
+`fundingContractAccountId`: The contract accountId that was used to send the funds.
+`linkdropKeyPairSecretKey`: The corresponding secret key to the public key sent to the contract.
+`redirectUrl`: The url that wallet will redirect to after funds are successfully claimed to an existing account. The URL is sent the accountId used to claim the funds as a query param.
+
+### Claiming a linkdrop without a NEAR Account
+
+The receiver of the link that is claiming the linkdrop:
+
+- Receives the link which redirects them to a linkdrop claiming page (such as NEAR wallet).
+- The linkdrop claim page calls `get_key_information` to display information associated with the linkdrop
+- The linkdrop claiming page creates a new keypair (`pubKey2`, `privKey2`) locally. (The keypair is not written to chain at this time)
+- Receiver will then choose an account ID such as alice.near.
+- Wallet will then use the privKey1 which has access to call `claim` and `create_account_and_claim` in order to call `create_account_and_claim` on the contract.
+- It will pass in `pubKey2` which will be used to create a full access key for the new account.
+- The contract will create the new account and transfer the funds to it alongside any NFT or fungible tokens included in the linkdrop.
+
+### Claiming a linkdrop with a NEAR Account
+The receiver of the link that is claiming the linkdrop:
+
+- Receives the link which redirects them to a linkdrop claiming page (such as NEAR wallet).
+- The linkdrop claim page calls `get_key_information` to display information associated with the linkdrop
+- The linkdrop claiming page connects to an existing NEAR Account.
+- Wallet will then use the `privKey1` which has access to call `claim` and `create_account_and_claim` in order to call `claim` on the contract.
+- The contract will transfer the funds to the existing account alongside any NFT or fungible tokens included in the linkdrop.
+
+
+## Reference Implementation
+
+Below are some references for linkdrop contracts. A specific implementation of this standard can be seen in [this pull request](https://github.com/near/near-linkdrop/pull/24).◊
+
+- [Link Drop Contract](https://github.com/near/near-linkdrop)
+
+- [Keypom Contract](https://github.com/keypom/keypom)
+
+
+## Security Implications
```

Comment:
Given that account creation and FT/NFT transfers happen in different calls, there should be some security model around error handling: rolling back or ignoring things.